code/go/0chain.net/chaincore/block/entity_test.go

```diff
@@ -0,0 +1,3645 @@
+		isNotarized           bool
+		ticketsMutex          *sync.RWMutex
+		verificationStatus    int
+		RunningTxnCount       int64
+		UniqueBlockExtensions map[string]bool
+		MagicBlock            *MagicBlock
+	}
+	type args struct {
+		prevBlock *Block
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   *Block
+	}{
+		{
+			name: "OK",
+			fields: fields{
+				UnverifiedBlockBody:   b.UnverifiedBlockBody,
+				VerificationTickets:   b.VerificationTickets,
+				HashIDField:           b.HashIDField,
+				Signature:             b.Signature,
+				ChainID:               b.ChainID,
+				ChainWeight:           b.ChainWeight,
+				RoundRank:             b.RoundRank,
+				PrevBlock:             b.PrevBlock,
+				TxnsMap:               b.TxnsMap,
+				ClientState:           b.ClientState,
+				stateStatus:           b.stateStatus,
+				blockState:            b.blockState,
+				isNotarized:           b.isNotarized,
+				verificationStatus:    b.verificationStatus,
+				RunningTxnCount:       b.RunningTxnCount,
+				UniqueBlockExtensions: b.UniqueBlockExtensions,
+				ticketsMutex:          b.ticketsMutex,
+				MagicBlock:            b.MagicBlock,
+			},
+			args: args{prevBlock: prevB},
+			want: func() *Block {
+				b := NewBlock("", 2)
+				b.PrevBlock = copyBlock(prevB)
+				b.PrevHash = prevB.Hash
+				b.Round = prevB.Round + 1
+				if len(b.PrevBlockVerificationTickets) == 0 {
+					b.PrevBlockVerificationTickets = prevB.GetVerificationTickets()
+				}
+
+				return b
+			}(),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &Block{
+				UnverifiedBlockBody:   tt.fields.UnverifiedBlockBody,
+				VerificationTickets:   tt.fields.VerificationTickets,
+				HashIDField:           tt.fields.HashIDField,
+				Signature:             tt.fields.Signature,
+				ChainID:               tt.fields.ChainID,
+				ChainWeight:           tt.fields.ChainWeight,
+				RoundRank:             tt.fields.RoundRank,
+				PrevBlock:             tt.fields.PrevBlock,
+				TxnsMap:               tt.fields.TxnsMap,
+				ClientState:           tt.fields.ClientState,
+				stateStatus:           tt.fields.stateStatus,
+				blockState:            tt.fields.blockState,
+				isNotarized:           tt.fields.isNotarized,
+				ticketsMutex:          tt.fields.ticketsMutex,
+				verificationStatus:    tt.fields.verificationStatus,
+				RunningTxnCount:       tt.fields.RunningTxnCount,
+				UniqueBlockExtensions: tt.fields.UniqueBlockExtensions,
+				MagicBlock:            tt.fields.MagicBlock,
+			}
+
+			b.SetPreviousBlock(tt.args.prevBlock)
+
+			// setting mutexes and states to nil because they are not comparable
+			nilBlocksMutexes(tt.want)
+			nilBlocksMutexes(b)
+
+			if !assert.Equal(t, tt.want, b) {
+				t.Errorf("SetPreviousBlock() got = %v, want = %v", b, tt.want)
+			}
+		})
+	}
+}
+
+func TestBlock_SetStateDB_Debug_True(t *testing.T) {
+	t.Skip("need to protect debug global variable against parallel access")
```

Comment:
No reason to skip. Moving `state.SetDebugLevel(1)` should fix any confounding race error.

Reply:
Moved and fixed.